commons/src/test/java/io/aiven/kafka/tiered/storage/commons/chunkindex/ChunkIndexSerializationTest.java

```diff
@@ -0,0 +1,133 @@
+import java.lang.reflect.Field;
+import java.nio.ByteBuffer;
+import java.util.Arrays;
+import java.util.Base64;
+import java.util.List;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.github.luben.zstd.ZstdCompressCtx;
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class ChunkIndexSerializationTest {
+    final ObjectMapper mapper = new ObjectMapper();
+
+    static final String ENCODED_CHUNKS = "KLUv/SAPeQAAAAAAAwAAAAoBAAoAAAAe";
+
+    // This should give an idea what's inside ENCODED_CHUNKS.
+    static {
+        // Three values are encoded: 10, 20, 30.
+        final ByteBuffer buf = ByteBuffer.allocate(100);
+        buf.putInt(3);  // count
+        buf.putInt(10);  // base
+        buf.put((byte) 1);  // bytes per value
+        buf.put((byte) 0);  // i.e. 10
+        buf.put((byte) 10);  // i.e. 20
+        buf.putInt(30);  // i.e. 30 (no base)
+        buf.limit(buf.position());
+        buf.rewind();
+
+        final byte[] compressed;
+        try (final ZstdCompressCtx compressCtx = new ZstdCompressCtx()) {
+            compressCtx.setContentSize(true);
+            compressed = compressCtx.compress(Arrays.copyOfRange(buf.array(), 0, buf.limit()));
+        }
+        final String expectedEncodedValue = Base64.getEncoder().encodeToString(compressed);
+        assertThat(expectedEncodedValue).isEqualTo(ENCODED_CHUNKS);
+    }
+
+    static final String FIXED_SIZE_CHUNK_INDEX_JSON = "{"
+        + "\"type\":\"fixed\","
+        + "\"originalChunkSize\":100,"
+        + "\"originalFileSize\":250,"
+        + "\"transformedChunkSize\":110,"
+        + "\"finalTransformedChunkSize\":30}";
+
+    static final String VARIABLE_CHUNK_INDEX_JSON = "{"
+        + "\"type\":\"variable\","
+        + "\"originalChunkSize\":100,"
+        + "\"originalFileSize\":250,"
+        + "\"transformedChunks\":\"" + ENCODED_CHUNKS + "\"}";
+
+    @Test
+    void serializeFixedSizeChunkIndex() throws JsonProcessingException {
+        final var fixedSizeChunkIndex = new FixedSizeChunkIndex(
+            100, 250, 110, 30);
+        assertThat(mapper.writeValueAsString(fixedSizeChunkIndex))
+            .isEqualTo(FIXED_SIZE_CHUNK_INDEX_JSON);
+    }
+
+    @Test
+    void deserializeFixedSizeChunkIndex() throws IOException, NoSuchFieldException, IllegalAccessException {
+        final FixedSizeChunkIndex index = mapper.readValue(FIXED_SIZE_CHUNK_INDEX_JSON, FixedSizeChunkIndex.class);
+
+        assertThat(index.originalChunkSize).isEqualTo(100);
+        assertThat(index.originalFileSize).isEqualTo(250);
+        assertThat(index.finalTransformedChunkSize).isEqualTo(30);
+
+        final Field field = index.getClass().getDeclaredField("transformedChunkSize");
+        field.setAccessible(true);
+        final int transformedChunkSize = (int) field.get(index);
+        assertThat(transformedChunkSize).isEqualTo(110);
```

Comment:
Why not just `index.transformedChunkSize`?

Reply:
Sorry, a leftover from times when the field was private. Fixed